<?xml version="1.0"?>
<launch>
    <arg name="sim" default="true"/>

    <param name="/use_sim_time" value="true" />

    <group ns="algorithm">
        <rosparam file="$(find dfc_mas_fr)/config/algorithm_params.yaml" />

        <node pkg="dfc_mas_fr" type="Publisher.py" name="publisher" output="screen" args="--sim --dt=0.05 --vis=mpl"/>
            
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node1"/>
            <arg name="id" value="1"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node2"/>
            <arg name="id" value="2"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node3"/>
            <arg name="id" value="3"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node4"/>
            <arg name="id" value="4"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node5"/>
            <arg name="id" value="5"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node6"/>
            <arg name="id" value="6"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node7"/>
            <arg name="id" value="7"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node8"/>
            <arg name="id" value="8"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node9"/>
            <arg name="id" value="9"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>
        <include file="$(find dfc_mas_fr)/launch/node.launch">
            <arg name="name" value="node10"/>
            <arg name="id" value="10"/>
            <arg name="sim" value="$(arg sim)"/>
        </include>

        
        <node pkg="dfc_mas_fr" type="MarkerPublisher.py" name="marker_publisher" output="screen" args="node"/>

        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dfc_mas_fr)/launch/rviz_cfg.rviz"/>
    </group>
</launch>